src/panels/lovelace/special-rows/hui-conditional-row.ts

```diff
@@ -0,0 +1,75 @@
+import { customElement } from "lit-element";
+
+import { createRowElement } from "../common/create-row-element";
+import {
+  checkConditionsMet,
+  validateConditionalConfig,
+} from "../../lovelace/common/validate-condition";
+import { HomeAssistant } from "../../../types";
+import { LovelaceRow, ConditionalRowConfig } from "../entity-rows/types";
+
+@customElement("hui-conditional-row")
+class HuiConditionalRow extends HTMLElement implements LovelaceRow {
+  private _hass?: HomeAssistant;
+  private _config?: ConditionalRowConfig;
+  private _row?: LovelaceRow;
+
+  public setConfig(config) {
+    if (!config.row) {
+      throw new Error("No row option configured.");
+    }
+
+    if (!config.conditions) {
+      throw new Error("No conditions option configured.");
+    }
+
+    if (!Array.isArray(config.conditions)) {
+      throw new Error("conditions option is not an array");
+    }
+
+    if (!validateConditionalConfig(config.conditions)) {
+      throw new Error("conditions option is invalid.");
+    }
+
+    if (this._row && this._row.parentElement) {
+      this.removeChild(this._row);
+    }
+
+    this._config = config;
+    this._row = createRowElement(config.row) as LovelaceRow;
+
+    this.update();
+  }
+
+  set hass(hass: HomeAssistant) {
+    this._hass = hass;
+
+    this.update();
```

Comment:
You can also use a `updatingElement` for these kind of things, I will do the checking if properties changed for you.

Reply:
Don't quite follow. This? https://lit-element.polymer-project.org/api/classes/_lib_updating_element_.updatingelement.html

Reply:
Yes, that one :-)

Reply:
Is there an advantage to that over what I'm doing?

Reply:
It looks more like a LitElement and you don't need setters.

Reply:
I'm not getting it and not seeing the advantage 🤷‍♂️ sorry. I'm missing something

Reply:
Hmm, moving to updatingElement caused errors
https://gist.github.com/iantrich/3910304ba65d0d86e39167eccc2a45a9
```
connection.ts:267 Uncaught (in promise) {code: "config_not_found", message: "No config found."}
./node_modules/home-assistant-js-websocket/dist/haws.es.js.c._handleMessage @ connection.ts:267
(anonymous) @ connection.ts:100
hass-router-page.ts:149 Error loading page lovelace TypeError: clazz.createProperty is not a function
    at finisher (decorators.ts:114)
    at Object.runClassFinishers (hui-cast-row.ts:23)
    at _decorate (hui-cast-row.ts:23)
    at Module../src/panels/lovelace/special-rows/hui-conditional-row.ts (hui-conditional-row.ts:12)
    at __webpack_require__ (bootstrap:63)
    at Module../src/panels/lovelace/common/create-row-element.ts (create-row-element.ts:1)
    at __webpack_require__ (bootstrap:63)
    at Module../src/panels/lovelace/cards/hui-entities-card.ts (hui-conditional-card.ts:73)
    at __webpack_require__ (bootstrap:63)
    at Module../src/panels/lovelace/common/create-card-element.ts (create-card-element.ts:1)
(anonymous) @ hass-router-page.ts:149
Promise.catch (async)
update @ hass-router-page.ts:147
performUpdate @ updating-element.ts:704
_enqueueUpdate @ updating-element.ts:653
async function (async)
_enqueueUpdate @ updating-element.ts:643
_requestUpdate @ updating-element.ts:605
set @ updating-element.ts:313
rebuild @ hass-router-page.ts:259
async function (async)
rebuild @ hass-router-page.ts:256
_updateRoutes @ partial-panel-resolver.ts:141
updated @ partial-panel-resolver.ts:108
performUpdate @ updating-element.ts:720
_enqueueUpdate @ updating-element.ts:653
async function (async)
_enqueueUpdate @ updating-element.ts:643
_requestUpdate @ updating-element.ts:605
initialize @ updating-element.ts:449
UpdatingElement @ updating-element.ts:438
HassRouterPage @ hass-router-page.ts:49
PartialPanelResolver @ partial-panel-resolver.ts:91
_clone @ template-instance.ts:97
__commitTemplateResult @ parts.ts:276
commit @ parts.ts:218
render @ render.ts:55
render @ shady-render.ts:277
update @ lit-element.ts:219
performUpdate @ updating-element.ts:704
_enqueueUpdate @ updating-element.ts:653
async function (async)
_enqueueUpdate @ updating-element.ts:643
_requestUpdate @ updating-element.ts:605
initialize @ updating-element.ts:449
initialize @ lit-element.ts:146
UpdatingElement @ updating-element.ts:438
LitElement @ lit-element.ts:61
HomeAssistantMain @ home-assistant-main.ts:36
_clone @ template-instance.ts:97
__commitTemplateResult @ parts.ts:276
commit @ parts.ts:218
update @ template-instance.ts:53
__commitTemplateResult @ parts.ts:268
commit @ parts.ts:218
render @ render.ts:55
render @ shady-render.ts:277
update @ lit-element.ts:219
performUpdate @ updating-element.ts:704
_enqueueUpdate @ updating-element.ts:653
async function (async)
_enqueueUpdate @ updating-element.ts:643
_requestUpdate @ updating-element.ts:605
set @ updating-element.ts:313
_updateHass @ hass-base-mixin.ts:49
(anonymous) @ connection-mixin.ts:142
n @ store.ts:37
(anonymous) @ collection.ts:29
Promise.then (async)
c @ collection.ts:28
u @ collection.ts:31
subscribe @ collection.ts:52
L @ services.ts:74
(anonymous) @ core.ts:80
Promise.then (async)
./src/entrypoints/core.ts @ core.ts:74
__webpack_require__ @ bootstrap:63
(anonymous) @ bootstrap:198
(anonymous) @ bootstrap:198
hass-router-page.ts:157 Uncaught (in promise) TypeError: Failed to execute 'removeChild' on 'Node': parameter 1 is not of type 'Node'.
    at hass-router-page.ts:157
```

Reply:
https://gist.github.com/iantrich/3910304ba65d0d86e39167eccc2a45a9#file-hui-conditional-row-ts-L12

Reply:
You should extend `updatingElement`

Reply:
That makes a little too much sense 😂